Generals/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5428,21 +5428,33 @@ void InGameUI::selectNextIdleWorker( void )
 	else
 	{
 		Drawable *selectedDrawable = TheInGameUI->getFirstSelectedDrawable();	
+		ObjectList uniqueIdleWorkers;
```

Comment:
To improve performance a bit, use `std::vector<Object*>` and `reserve(idleWorkers.size())`